### PR TITLE
Electron based wallet

### PR DESCRIPTION
**Pull request by gabrielbazan7**
- [x] Chrome-storage to file-storage
- [x] Copy to clipboard
- [x] Open a new window
- [x] Open link to an external window
- [x] Open url (deep link) -> osx
- [x] Open url (deep link) -> windows (done but needs to be tested)
- [x] Open url (deep link) -> linux (done but needs to be tested)
- [x] Determine if platform is desktop or whatever
- [x] Make camera works
- [x] Import file backup
- [x] Check update desktop on focus events
- [x] Test Coinbase and Glidera
- [x] Create exe and setup installer
- [x] Migration ( no migration )
- [x] Build process -> snap for Linux, appx for Windows, mas for Mac

### CODE REVIEW

File: electron/main.js [hunk 1]
```diff
@@ -26,24 +39,52 @@ function createWindow() {
...
+app.setAsDefaultProtocolClient('bitpay');
+app.setAsDefaultProtocolClient('bitcoin');
 app.setVersion(appConfig.version);
 app.setName(appConfig.nameCase);
 
-let nameFolder = appConfig.name + '-v4';
+const getHomeDirPath = platform => {
+  switch (platform) {
+    case 'win32':
+      return process.env['USERPROFILE'];
```
Comment (matiu) [inline electron/main.js]: if the user change "userprofile" variable, the will no be able to find his wallets, right?

Comment (matiu) [reply electron/main.js]: can't we use a `homedir()` thing? if not, add this to the README or something

File: electron/main.js [hunk 2]
```diff
@@ -26,24 +39,52 @@ function createWindow() {
...
+  win.webContents.on('did-finish-load', () => {
+    // Windows: Handle deeplink url
+    if (process.platform == 'win32') {
+      // Keep only command line / deep linked arguments
+      deeplinkingUrl = process.argv.slice(1);
+    }
+    if (deeplinkingUrl) {
+      setTimeout(() => {
+        win.webContents.send('open-url-event', deeplinkingUrl);
+      }, 1000);
```
Comment (matiu) [inline electron/main.js]: shouldn't the event 'did-finish-load' be the right timing to send 'open-url-event' ? Why the extra 1000 ms?

File: src/pages/integrations/coinbase/coinbase.ts [hunk 3]
```diff
@@ -130,28 +130,30 @@ export class CoinbasePage {
...
+        webPreferences: { nodeIntegration: false }
+      });
+      win.once('ready-to-show', () => {
+        win.show();
+        win.focus();
+      });
+      win.loadURL(oauthUrl);
+      win.webContents.on('did-finish-load', () => {
+        const title = win.webContents.getTitle();
+        if (title.indexOf('Coinbase') == -1) {
```
Comment (matiu) [inline src/pages/integrations/coinbase/coinbase.ts]: lets use something different than the tile to identify the window here. Maybe add some metadata or an extra attribute.

File: electron/main.js (hunk 1 above)
Comment (gabrielbazan7) [reply electron/main.js]: Yes, right

File: src/providers/persistence/persistence.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export class PersistenceProvider {
   public load() {
     this.storage = this.platform.isCordova
       ? new FileStorage(this.file, this.logger)
-      : new LocalStorage(this.platform, this.logger);
+      : new LocalStorage(this.logger);
```
Comment (matiu) [inline src/providers/persistence/persistence.ts]: Can we test that Electron support >1MB data per app?

File: src/providers/persistence/storage/local-storage.ts [hunk 5]
```diff
@@ -1,58 +1,17 @@
...
 export class LocalStorage implements IStorage {
   ls;
 
-  constructor(
-    private platformProvider: PlatformProvider,
-    private logger: Logger
-  ) {
+  constructor(private logger: Logger) {
     this.ls =
       typeof window.localStorage !== 'undefined' ? window.localStorage : null;
```
Comment (matiu) [inline src/providers/persistence/storage/local-storage.ts]: lets simplify this:
```
if (!window.localStorage)
  throw new Error('...');
```
ls = window.localStorage;

File: src/providers/persistence/storage/local-storage.ts [hunk 6]
```diff
@@ -69,43 +28,23 @@ export class LocalStorage implements IStorage {
...
 
   remove(k: string): Promise<void> {
     return new Promise<void>(resolve => {
-      if (this.platformProvider.isNW) {
-        chrome.storage.local.remove(k);
-      } else {
-        this.ls.removeItem(k);
-      }
+      this.ls.removeItem(k);
       this.logger.debug('File removed: ' + k);
```
Comment (matiu) [inline src/providers/persistence/storage/local-storage.ts]: Shoiuld be "Storage Key <> removed" (not file)

File: src/providers/persistence/storage/local-storage.ts [hunk 7]
```diff
@@ -69,43 +28,23 @@ export class LocalStorage implements IStorage {
...
-        chrome.storage.local.get(k, data => {
-          let v = data[k];
-          return resolve(this.processingData(v));
-        });
-      } else {
-        let v = this.ls.getItem(k);
-        return resolve(this.processingData(v));
-      }
+      let v = this.ls.getItem(k);
+      return resolve(this.processingData(v));
```
Comment (matiu) [inline src/providers/persistence/storage/local-storage.ts]: please rename `processingData(v)` to `processValue(v)`;

File: electron/main.js (hunk 2 above)
Comment (gabrielbazan7) [reply electron/main.js]: The problem here is that is not yet subscribed to the 'open-url-event' event, this takes place after the profile loaded. And we need the profile loaded and all the wallets to handle de URL correctly.

File: electron/main.js [hunk 8]
```diff
@@ -26,24 +39,52 @@ function createWindow() {
...
+    }
+  });
 }
 
+app.setAsDefaultProtocolClient('copay');
+app.setAsDefaultProtocolClient('bitpay');
+app.setAsDefaultProtocolClient('bitcoin');
```
Comment (cmgustavo) [inline electron/main.js]: Missing `bitcoincash:` and `bchtest`?

Comment (gabrielbazan7) [reply electron/main.js]: done. Thanks

File: src/providers/persistence/persistence.ts (hunk 4 above)
Comment (gabrielbazan7) [reply src/providers/persistence/persistence.ts]: I did a manual test running a script and it supports up to 5MB.

File: src/pages/integrations/coinbase/coinbase.ts (hunk 3 above)
Comment (gabrielbazan7) [reply src/pages/integrations/coinbase/coinbase.ts]: Currently, that is the easiest way to identify that the title has the auth code. But, we could plan a refactor for later.

**Pull request merged**